src/sentry/monitors/tasks.py

```diff
@@ -392,3 +411,58 @@ def mark_checkin_timeout(checkin_id: int, ts: datetime, **kwargs):
         )
 
         mark_failed(checkin, ts=most_recent_expected_ts)
+
+
+@instrumented_task(
+    name="sentry.monitors.tasks.detect_broken_monitor_envs",
+    max_retries=0,
+    time_limit=15 * 60,
+    soft_time_limit=10 * 60,
+    record_timing=True,
+)
+def detect_broken_monitor_envs():
+    current_time = django_timezone.now()
+    open_incidents_qs = MonitorIncident.objects.select_related("monitor").filter(
+        resolving_checkin=None,
+        starting_timestamp__lte=(current_time - timedelta(days=NUM_DAYS_BROKEN_PERIOD)),
+        # TODO(davidenwang): When we want to email users, remove this filter
+        monitorenvbrokendetection__isnull=True,
+    )
+    for open_incidents in chunked(
+        RangeQuerySetWrapper(
+            open_incidents_qs,
+            step=1000,
+        ),
```

Comment:
Let's also include `order_by=starting_timestamp` here. Otherwise it has to order the whole set of incidents by id separately, rather than just using the index

Reply:
makes sense, also this comment confused me a little bit https://github.com/getsentry/sentry/blob/4454c16924b310d03584a1f85e4b2114ff53070b/src/sentry/utils/query.py#L81, is this referring to a different kind of usage of order_by?